dot/parachain/statement-distribution/cluster_tracker.go

```diff
@@ -408,3 +408,67 @@ func (c *clusterTracker) validatorSeconded(
 	}
 	return false
 }
+
+// canSend queries whether we can send a statement to a given validator.
+func (c *clusterTracker) canSend(
+	target parachaintypes.ValidatorIndex,
+	originator parachaintypes.ValidatorIndex, //nolint:unparam
+	statement parachaintypes.CompactStatement,
+) acceptOrRejectOutgoing {
+	if !c.isInGroup(target) || !c.isInGroup(originator) {
+		return notInGroupOutgoing{}
+	}
+
+	if c.theyKnowStatement(target, originator, statement) {
+		return knownOutgoing{}
+	}
+
+	switch statement.(type) {
+	case *parachaintypes.CompactSeconded:
+		// we send the same `Seconded` statements to all our peers, and only the first `k`
+		// from each originator.
+		if !c.secondedAlreadyOrWithinLimit(originator, statement.CandidateHash()) {
+			return excessiveSecondedOutgoing{}
+		}
+		return ok{}
+	case *parachaintypes.CompactValid:
+		if !c.knowsCandidate(target, statement.CandidateHash()) {
+			return candidateUnknownOutgoing{}
+		}
+		return ok{}
+	default:
+		panic("unreachable")
+	}
+}
+
+// noteSent notes that we sent an outgoing statement to a peer in the group.
+// This must be preceded by a successful `can_send` call.
+func (c *clusterTracker) noteSent(
+	target parachaintypes.ValidatorIndex,
+	originator parachaintypes.ValidatorIndex,
+	statement parachaintypes.CompactStatement,
+) {
+	targetKnowledge, ok := c.knowledge[target]
+	if !ok {
+		targetKnowledge = map[taggedKnowledge]struct{}{
+			outgoingP2P{specific{statement, originator}}: {},
+		}
+		c.knowledge[target] = targetKnowledge
+	}
+
+	if _, ok := statement.(*parachaintypes.CompactSeconded); ok {
```

Comment:
Again the same question.
Why it's a pointer?
should not it be
`statement.(parachaintypes.CompactSeconded)`

Reply:
yeah,m you are right.